engine/src/test/java/io/zeebe/engine/processor/workflow/subprocess/EventSubprocessTest.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.subprocess;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+
+import io.zeebe.engine.util.EngineRule;
+import io.zeebe.model.bpmn.Bpmn;
+import io.zeebe.model.bpmn.BpmnModelInstance;
+import io.zeebe.model.bpmn.builder.EmbeddedSubProcessBuilder;
+import io.zeebe.model.bpmn.builder.ProcessBuilder;
+import io.zeebe.protocol.record.Assertions;
+import io.zeebe.protocol.record.Record;
+import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import io.zeebe.protocol.record.value.BpmnElementType;
+import io.zeebe.protocol.record.value.WorkflowInstanceRecordValue;
+import io.zeebe.test.util.record.RecordingExporter;
+import io.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.List;
+import java.util.Map;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class EventSubprocessTest {
```

Comment:
We need more test cases to verify the behavior of event subprocesses. Instead of writing them now, we can also do this later.

Verify that
* the workflow instance stays active if one event subprocess instance is active
* the instance of the event subprocess is terminated on scope termination
* the event subprocess stays active until all (concurrent) inner instances are completed
* the event subprocess can be created multiple times
* all event subprocesses of a workflow can be created
* the event subscription of the event subprocess is closed when the scope is completed/terminated
* output mappings are applied when the event subprocess is created (for message start events)
* variables of the event subprocess are not propagated to the scope instance